resources/sts/4.14/hypershift/sts_extended_hcp_support_permission_policy.json

```diff
@@ -62,49 +137,300 @@
+            "Sid": "AddressAttribute",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeAddressesAttribute"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "Addresses",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeAddresses"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "AccountAttributes",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeAccountAttributes"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "ByoipCidrs",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeByoipCidrs"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "CapacityReservation",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeCapacityReservations"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "CarrierGateways",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeCarrierGateways"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "Instances",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeClassicLinkInstances"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "CustomerGateways",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeCustomerGateways"
+            ],
             "Resource": "*"
         },
         {
+            "Sid": "InstanceTypeOfferings",
             "Effect": "Allow",
             "Action": [
-                "s3:ListBucket"
+                "ec2:DescribeInstanceTypeOfferings"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "KeyPairs",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeKeyPairs"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "LaunchTemplate",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeLaunchTemplates"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "PlacementGroups",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribePlacementGroups"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "elb",
+            "Effect": "Allow",
+            "Action": [
+                "elasticloadbalancing:ConfigureHealthCheck"
+            ],
+            "Resource": "arn:aws:elasticloadbalancing:*:*:loadbalancer/*"
+        },
+        {
+            "Sid": "kms",
+            "Effect": "Allow",
+            "Action": [
+                "kms:CreateGrant"
+            ],
+            "Resource": "arn:aws:kms:*:*:key/*"
```

Comment:
This is too permissive. Please match the restrictions in the managed policies for KMS key usage.

Reply:
added condition 
`
"Condition": {
          "StringEquals": {
            "aws:ResourceTag/red-hat": "true"
          }
`